xdsl/dialects/llvm.py

```diff
@@ -396,18 +396,96 @@ def print(self, printer: Printer) -> None:
         printer.print(self.lhs.type)
 
 
+class OverflowFlag(StrEnum):
+    NO_SIGNED_WRAP = "nsw"
+    NO_UNSIGNED_WRAP = "nuw"
+
+
+@dataclass(frozen=True, init=False)
+class OverflowAttrBase(BitEnumAttribute[OverflowFlag]):
+    none_value = "none"
+
+
+@irdl_attr_definition
+class OverflowAttr(OverflowAttrBase):
+    name = "llvm.overflow"
+
+
+class ArithmeticBinOpOverflow(Generic[ArgT], IRDLOperation, ABC):
+    """Class for arithmetic binary operations that use overflow flags."""
+
+    T = Annotated[Attribute, ArgT, ConstraintVar("T")]
+
+    lhs = operand_def(T)
+    rhs = operand_def(T)
+    res = result_def(T)
+    overflowFlags = prop_def(OverflowAttr)
+
+    traits = frozenset([NoMemoryEffect()])
+
+    def __init__(
+        self,
+        lhs: SSAValue,
+        rhs: SSAValue,
+        attributes: dict[str, Attribute] = {},
+        overflow: OverflowAttr = OverflowAttr(None),
+    ):
+        super().__init__(
+            operands=[lhs, rhs],
+            attributes=attributes,
+            result_types=[lhs.type],
+            properties={
+                "overflowFlags": overflow,
+            },
+        )
+
+    @classmethod
+    def parse_overflow(cls, parser: Parser) -> OverflowAttr:
+        if parser.parse_optional_keyword("overflow") is not None:
+            return OverflowAttr(OverflowAttr.parse_parameter(parser))
+        return OverflowAttr("none")
+
+    def print_overflow(self, printer: Printer) -> None:
+        if self.overflowFlags.flags:
+            printer.print(" overflow")
+            self.overflowFlags.print_parameter(printer)
+
+    @classmethod
+    def parse(cls, parser: Parser):
+        lhs = parser.parse_unresolved_operand()
+        parser.parse_characters(",")
+        rhs = parser.parse_unresolved_operand()
+        overflowFlags = cls.parse_overflow(parser)
+        attributes = parser.parse_optional_attr_dict()
+        if "overflowFlags" in attributes:
+            flags = attributes.pop("overflowFlags")
+            if isinstance(flags, OverflowAttr):
+                overflowFlags = flags
```

Comment:
What's MLIR's behaviour here?

Reply:
I don't know
And I'm not sure how to check for that